r2/r2/public/static/js/reddit.js

```diff
@@ -1229,6 +1229,17 @@ $(function() {
             $("#moresearchinfo").slideUp();
             event.preventDefault();
         });
+
+        /* Add a scheme to URL input fields if not already present */
+        $("input[type=url]").blur(function() {
+            var input = $(this).val();
+            if (input && input.length > 0) {
+                var schemePresent = input.match(/^[a-z]+:\/\/.*/i);
```

Comment:
Using [the RegExp constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp) would get rid of those nasty backslash escapes:

    RegExp('^[a-z]+://.*', 'i')